src/main/java/mini/board/domain/login/LoginService.java

```diff
@@ -15,10 +17,20 @@ public LoginService(UserService userService) {
         this.userService = userService;
     }
 
-    public Optional<User> signIn(User user) {
-        return Optional.ofNullable(userService.findByLoginId(user)
-                .filter(m -> m.getPassword().equals(user.getPassword()))
-                .orElse(null));
+    public User signIn(User user) {
+
+        Optional<User> getUser = userService.findByLoginId(user);
+
+        if (getUser.isEmpty()) {
+            throw new APIError("EmptyLoginId", "아이디가 존재하지 않습니다.");
+        }
+        if (!getUser.get().getPassword().equals(user.getPassword())) {
+            throw new APIError("InconsistencyPassword", "비밀번호가 일치하지 않습니다.");
+        }
+
+        return getUser.get();
```

Comment:
optional 은 값이 있거나 없거나 입니다.

이 두 상태를 항상 생각해야 합니다. 상태에 따라 분기가 발생할 수 있죠.
이 분기를 염두 하는 건 코드를 이해하기 힘들고 실수할 여지가 있습니다.

앞서 "존재하다"를 검증했다면 뒤에는 "존재하다"를 검증할 필요가 없습니다.
아래와 같이 바꿀 수 있을 거 같네요.

```suggestion
        User getUser = userService.findByLoginId(user).orElse(Optional.empty());

        if (getUser == null) {
            throw new APIError("EmptyLoginId", "아이디가 존재하지 않습니다.");
        }
        if (!getUser.getPassword().equals(user.getPassword())) {
            throw new APIError("InconsistencyPassword", "비밀번호가 일치하지 않습니다.");
        }

        return getUser;
```